cloudmarker/clouds/gcpcloud.py

```diff
@@ -226,3 +192,112 @@ def done(self):
         up tasks associated with the :class:`GCPCloud` plugin. This
         may change in future.
         """
+
+
+def _get_normalized_firewall_rules(firewall):
+    """Split a firewall record into multiple firewall rules.
+
+    A firewall record in GCP has a top-level key named either
+    ``allowed`` or ``denied``. The value of this key is a list of
+    allowed or denied rules. Each such rule contains the name of allowed
+    or denied protocol along with a list of allowed or denied port
+    ranges.
+
+    In order to make it easier to write event plugins to detect security
+    issues in firewall, we generate a new firewall rule record for each
+    allowed or denied rule we find in the value of ``allowed`` or
+    ``denied`` keys in a firewall record.
+
+    Arguments:
+        firewall (dict): Raw firewall record retrieved from GCP.
+    """
+    allow_rules = firewall.get('allowed')
+    if allow_rules is not None:
+        for rule in allow_rules:
+            firewall_rule = _get_normalized_firewall_rule(firewall, rule)
+            firewall_rule['com']['access'] = 'allow'
+            yield firewall_rule
+
+    deny_rules = firewall.get('denied')
+    if deny_rules is not None:
+        for rule in deny_rules:
+            firewall_rule = _get_normalized_firewall_rule(firewall, rule)
+            firewall_rule['com']['access'] = 'allow'
```

Comment:
Why the `access` for `deny_rules` is also marked as `allow`?

Reply:
This is a mistake. Thanks for noticing this. I will update the pull request and fix it.

Reply:
Fixed it at [line 230](https://github.com/cloudmarker/cloudmarker/pull/65/files#diff-c403e664c38d6edc528237cc5e434e05R230).

Also added an assert for it in the tests at [line 208](https://github.com/cloudmarker/cloudmarker/pull/65/files#diff-831ed9aecd6b6e0d34f3fb816425070bR208).